internal/api/websocket.go

```diff
@@ -76,7 +76,11 @@ func (cr *codeOceanToRawReader) readInputLoop(ctx context.Context) {
 		case <-readMessage:
 		}
 
-		if err != nil {
+		if err != nil && websocket.IsCloseError(err, websocket.CloseNormalClosure) {
+			log.Debug("ReadInputLoop: The client closed the connection!")
+			// The close handler will do something soon.
```

Comment:
We are experiencing a race condition here, right?

Reply:
Yes

Reply:
Alright. I think this handling is fine.